python/spglib/spglib.py

```diff
@@ -1215,28 +1219,76 @@ def get_magnetic_spacegroup_type(uni_number) -> dict | None:
     """
     _set_no_error()
 
-    keys = (
-        "uni_number",
-        "litvin_number",
-        "bns_number",
-        "og_number",
-        "number",
-        "type",
-    )
     msg_type_list = _spglib.magnetic_spacegroup_type(uni_number)
-    _set_error_message()
 
     if msg_type_list is not None:
-        msg_type = dict(zip(keys, msg_type_list))
-        for key in msg_type:
-            if key not in ["uni_number", "litvin_number", "number", "type"]:
-                msg_type[key] = msg_type[key].strip()
+        msg_type = MagneticSpaceGroupType(
+            uni_number=msg_type_list[0],
+            litvin_number=msg_type_list[1],
+            bns_number=msg_type_list[2].strip(),
+            og_number=msg_type_list[3].strip(),
+            number=msg_type_list[4],
+            type=msg_type_list[5],
+        )
+        return msg_type
+    else:
+        _set_error_message()
+        return None
+
+
+def get_magnetic_spacegroup_type_from_symmetry(
+    rotations: NDArray[np.intc],
+    translations: NDArray[np.double],
+    time_reversals: NDArray[np.intc],
+    lattice: NDArray[np.double] | None = None,
+    symprec: float = 1e-5,
+) -> MagneticSpaceGroupType | None:
+    """Return magnetic space-group type information from symmetry operations.
+
+    Parameters
+    ----------
+    rotations, translations, time_reversals:
+        See returns of :func:`get_magnetic_symmetry`.
+    lattice : (Optional) array_like (3, 3)
+        Basis vectors a, b, c given in row vectors. This is used as the measure of
+        distance. Default is None, which gives unit matrix.
+    symprec: float
+        See :func:`get_symmetry`.
+
+    Returns
+    -------
+    magnetic_spacegroup_type: MagneticSpaceGroupType
+        See :class:`MagneticSpaceGroupType` for the description of the keys.
```

Comment:
```suggestion
    magnetic_spacegroup_type: MagneticSpaceGroupType
```

It is a proper `class` so I don't think the comment is relevant anymore.